Polyhedron/demo/Polyhedron/include/CGAL/IO/read_surf_trianglemesh.h

```diff
@@ -101,10 +100,12 @@ void read_surf(std::istream& input, std::vector<Mesh>& output, std::vector<Mater
     }
   std::cout<<nb_patches<<" patch(es)"<<std::endl;
   metadata.resize(nb_patches);
-  for(int i=0; i < nb_patches; ++i)
+  output.resize(nb_patches);
+  const vertex_descriptor null_vertex = boost::graph_traits<Mesh>::null_vertex();
+  std::vector<vertex_descriptor> vertices(nb_vertices, null_vertex);
+  for(int i=0; i < nb_patches; /* i is incremented in the body */)
   {
-    std::map<std::size_t, vertex_descriptor> vertices;
-    Mesh mesh;
+    Mesh& mesh = output[i];
```

Comment:
The line 
```C++
  output.push_back(mesh);
```
is catastrophic: the vector may get reallocated, and all its content copied/moved. And a `Mesh` can be a huge surface, let's not copy it.

Instead, as soon as we get `nb_patches`, let's resize the vector (it is filled with empty `Mesh`es), and then line 108 `mesh ` is a ref to the ith mesh of the `output` vector, and is filled in-place. That's speed-up a lot!